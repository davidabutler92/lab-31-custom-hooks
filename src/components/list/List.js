import React from 'react';

import ListItem from '../listItem/ListItem';

export default function List() {
  return (
    <>
      <h1>List Page</h1>
      <ListItem />
    </>
  );
}
