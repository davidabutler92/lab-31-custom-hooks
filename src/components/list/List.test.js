import React from 'react';
import { render, cleanup } from '@testing-library/react';
import List from './List';
import { MemoryRouter } from 'react-router-dom';

const characters = [
  {
    id: 1,
    name: 'someName',
    image: 'something',
  },
  {
    id: 2,
    name: 'someName',
    image: 'somePhoto',
  },
  {
    id: 3,
    name: 'someName',
    image: 'somePhoto',
  },
];

describe('ListPage component', () => {
  afterEach(() => cleanup());
  it('renders list of characters', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <List characters={characters} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
