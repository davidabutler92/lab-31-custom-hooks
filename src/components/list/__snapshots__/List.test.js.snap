// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPage component renders list of characters 1`] = `
<DocumentFragment>
  <ul
    data-testid="characters"
  >
    <li>
      <a
        href="/detail/1"
      >
        <img
          alt="someName"
          src="something"
        />
        <h2>
          someName
        </h2>
      </a>
    </li>
    <li>
      <a
        href="/detail/2"
      >
        <img
          alt="someName"
          src="somePhoto"
        />
        <h2>
          someName
        </h2>
      </a>
    </li>
    <li>
      <a
        href="/detail/3"
      >
        <img
          alt="someName"
          src="somePhoto"
        />
        <h2>
          someName
        </h2>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
